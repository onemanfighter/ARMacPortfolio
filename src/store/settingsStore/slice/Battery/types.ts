export type BatteryState = {};

export interface BatterySettingsAction {}

export type BatteryStateSlice = BatteryState & BatterySettingsAction;
