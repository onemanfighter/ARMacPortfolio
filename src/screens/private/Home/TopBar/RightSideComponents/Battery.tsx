import { BatteryIcon, MacIcon, SearchIcon } from '@assets';
import {
  Box,
  Menu,
  MenuDivider,
  MenuItem,
  MenuList,
  Text,
} from '@chakra-ui/react';
import {
  MenuItemComponent,
  MenuListComponent,
  TopBarButton,
} from '@components';
import { useTranslation } from 'react-i18next';

const Battery = () => {
  const batteryPercentage = 67;
  const { t } = useTranslation();
  return (
    <Menu>
      <TopBarButton
        text=""
        onClick={() => {}}
        ariaLabel="battery-top-bar-button"
        icon={
          <BatteryIcon
            percentage={batteryPercentage}
            props={{
              width: '2em',
              height: '2em',
              color: 'white',
            }}
          />
        }
      />
      <MenuListComponent>
        <Box
          display={'flex'}
          justifyContent={'space-between'}
          flexDirection={'row'}
          mx={2}
        >
          <Text color={'white'} fontSize={'xs'} fontWeight={'bold'}>
            {t('TopAppBar.battery.title')}
          </Text>
          <Text color={'white'} fontSize={'xs'}>
            {batteryPercentage + '%'}
          </Text>
        </Box>
        <MenuDivider mx={2} />
        <MenuItemComponent
          text={t('TopAppBar.battery.batterySettings')}
          ariaLabel="new-tab"
          onClick={() => {}}
        />
      </MenuListComponent>
    </Menu>
  );
};

export default Battery;
