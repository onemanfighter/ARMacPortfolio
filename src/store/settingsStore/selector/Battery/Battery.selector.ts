import { SettingsStoreState } from '../../settingsStore';

const batterySelector = (state: SettingsStoreState) => ({});

export { batterySelector };
