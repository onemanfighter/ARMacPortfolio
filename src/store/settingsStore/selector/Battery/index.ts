export { batterySelector } from './Battery.selector';
