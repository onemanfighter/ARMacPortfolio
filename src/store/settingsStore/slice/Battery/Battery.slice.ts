import { SettingsStoreSlice } from '../../settingsStore';
import { BatteryState, BatteryStateSlice } from './types';

const defaultBatteryState: BatteryState = {};

const createBatterySlice: SettingsStoreSlice<BatteryStateSlice> = (set) => ({
  ...defaultBatteryState,
});

export default createBatterySlice;
