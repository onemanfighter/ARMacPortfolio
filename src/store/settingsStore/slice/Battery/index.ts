export { default as createBatterySlice } from './Battery.slice';
export type { BatteryStateSlice, BatteryState } from './types';
