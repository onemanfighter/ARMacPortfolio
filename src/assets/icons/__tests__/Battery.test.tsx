import { render } from '@testing-library/react';
import { BatteryIcon } from '../Battery';

describe('BatteryIcon', () => {
  it('should render correctly', () => {
    const { container } = render(<BatteryIcon />);

    expect(container).toMatchSnapshot();
  });
});
