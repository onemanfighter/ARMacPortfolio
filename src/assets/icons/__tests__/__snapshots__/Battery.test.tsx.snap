// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BatteryIcon should render correctly 1`] = `
<div>
  <svg
    fill="currentColor"
    height="1em"
    viewBox="0 0 16 16"
    width="1em"
  >
    <path
      d="M2 6h5v4H2V6z"
    />
    <path
      d="M2 4a2 2 0 00-2 2v4a2 2 0 002 2h10a2 2 0 002-2V6a2 2 0 00-2-2H2zm10 1a1 1 0 011 1v4a1 1 0 01-1 1H2a1 1 0 01-1-1V6a1 1 0 011-1h10zm4 3a1.5 1.5 0 01-1.5 1.5v-3A1.5 1.5 0 0116 8z"
    />
  </svg>
</div>
`;
